File: vue-zayit/Zayit-cs/Zayit/Models/Book.cs
namespace Zayit.Models
{
    public class Book
    {
        public int Id { get; set; }

        public int CategoryId { get; set; }

        public string Path { get; set; }

        public string Title { get; set; }

        public string HeShortDesc { get; set; }

        public int OrderIndex { get; set; }

        public int TotalLines { get; set; }

        public int HasTargumConnection { get; set; }

        public int HasReferenceConnection { get; set; }

        public int HasCommentaryConnection { get; set; }

        public int HasOtherConnection { get; set; }
    }

}

File: vue-zayit/zayit-vue/src/data/types/Book.ts
export interface Book {
    id: number
    categoryId: number
    title: string
    heShortDesc: string | null
    path?: string
    orderIndex: number
    totalLines: number
    hasSourceConnection: number
    hasTargumConnection: number
    hasCommentaryConnection: number
    hasReferenceConnection: number
    hasOtherConnection: number
    defaultCommentatorBookId: number | null
    period?: string // Category period (chronological): תנ"ך, ספרות חז"ל, גאונים, ראשונים, אחרונים, קבלה, מוסר וחסידות, הלכה, אחר
    pubDate?: string | null // Publication date from pub_date table
}

export function hasConnections(book: Book): boolean {
    return book.hasTargumConnection > 0 ||
        book.hasReferenceConnection > 0 ||
        book.hasCommentaryConnection > 0 ||
        book.hasOtherConnection > 0 ||
        book.hasSourceConnection > 0;
}

export interface SearchResult {
    book: Book
    matchType: 'title' | 'description'
}
